api/src/main/java/org/datacleaner/api/UsageMeteringMessage.java

```diff
@@ -53,12 +71,22 @@ public String getType() {
     /**
      * Additional info, which sematic is defined by the message producer
      * and should be understandable by consumer.
+     * Contains a CSV-formatted line with detail fields.
      */
-    public Map<String, String> getAdditionalInfo() {
-        return additionalInfo;
+    public String getDetails() {
+        return details;
     }
 
     public String toString() {
-        return "UsageMeteringMessage[" + type + ", " + additionalInfo + "]";
+        return "UsageMeteringMessage[" + type + " " + details+ "]";
     }
+
+    /** Escapes value for a CSV line */
+    private String escapeValue(String field) {
+        if(field.indexOf(DETAILS_SEPARATOR_CHAR) == -1 && field.indexOf(DETAILS_QUOTE_CHAR) == -1) {
+            return field;
+        }
+        return DETAILS_QUOTE_CHAR + field.replace(DETAILS_QUOTE_STR, DETAILS_TWO_QUOTES_STR) + DETAILS_QUOTE_CHAR;
```

Comment:
String.replace only replaces one time. We should probably use replaceAll to avoid multiple things to be escaped.